distribution/docs/src/main/resources/_contents/_endpoints/cometd-endpoint-contents.adoc

```diff
@@ -0,0 +1,845 @@
+----
+{
+  "cql":"(\"anyText\" ILIKE 'foo')",
+  "id":"3b19bc9c-2155-4ca6-bae8-65a9c8e373f6"
+}
+----
+
+This would return any results matching the text `foo` on the `/3b19bc9c-2155-4ca6-bae8-65a9c8e373f6` channel
+
+
+====== CometD Notifications
+
+Notifications are messages that are sent to clients to inform them of some significant event happening.
+Clients must subscribe to a notification channel to receive these messages.
+
+Notifications are published by the server on several notification channels depending on the type.
+
+* subscribing to ``/${ddf-branding-lowercase}/notifications/**`` will cause the client to receive all notifications.
+* subscribing to `/${ddf-branding-lowercase}/notifications/catalog/downloads` will cause the client to only receive notifications of downloads.
+
+====== Using CometD Notifications
+
+[NOTE]
+====
+The ${branding} Search UI serves as a reference implementation of how clients can use notifications.
+====
+
+Notifications are currently being utilized in the Catalog application for resource retrieval.
+When a user initiates a resource retrieval, the channel `/ddf/notification/catalog/downloads` is opened, where notifications indicating the progress of that resource download are sent.
+Any client interested in receiving these progress notifications must subscribe to that channel.
+${branding} starts downloading the resource to the client that requested it, a notification with a status of "Started" will be broadcast.
+If the resource download fails, a notification with a status of "Failed" will be broadcast.
+Or, if the resource download is being attempted again after a failure, "Retry" will be broadcast.
+When a notification is received, ${branding} Search UI displays a popup containing the contents of the notification, so a user is made aware of how their downloads are proceeding.
+Behind the scenes, the ${branding} Search UI invokes the REST endpoint to retrieve a resource.
+In this request, it adds the query parameter "user" with the CometD session ID or the unique User ID as the value.
+This allows the CometD server to know which subscriber is interested in the notification.
+For example, `http://${branding}_HOST:8181/services/catalog/sources/${branding-lowercase}.
+distribution/2f5db9e5131444279a1293c541c106cd?
+transform=resource&user=1w1qlo79j6tscii19jszwp9s2i55` notifications contain the following information:
+
+.Notification Contents
+[cols="1m,3,1" options="header"]
+|===
+|Parameter Name
+|Description
+|Required by ${branding} Search UI
+
+|application
+|"Downloads" for resource retrieval. This is used as a "type" or category of messages.
+|Yes
+
+|title
+|Resource/file name for resource retrieval.
+|Yes
+
+|message
+|Human-readable message containing status and a more detailed message.
+|Yes
+
+|timestamp
+|Timestamp in milliseconds of when event occurs.
+|Yes
+
+|user
+|CometD Session ID or unique User ID.
+|Yes
+
+|status
+|Status of event.
+|No
+
+|option
+|Resource retrieval option.
```

Comment:
We should add more information about what the options can be. @oconnormi might know what they are.